### PR TITLE
Kqueen ldap

### PR DESCRIPTION
**Pull request by naumvd95**
- refactor env-vars arch to provide configuration using only docker env-vars
- remove apply_env changes due issues with bad overriding in case of multiple value-entries in config file
- logging increased
- splitted on separate commits to increase visibility

p.s. need to refactor tests, related to config-file defining

### CODE REVIEW

File: kqueen/config/auth.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from kqueen.config import current_config
+
+
+class AuthModules():
```
Comment (atengler) [inline kqueen/config/auth.py]: Is there a reason to create this class?

File: kqueen/config/utils.py [hunk 2]
```diff
@@ -37,11 +36,14 @@ def apply_env_changes(config, prefix='KQUEEN_'):
         config (obj): Connfiguration object. This configuration will updated.
         prefix (str): Prefix for environment variables. Defaults to `KQUEEN_`.
 
+    return config_file
     """
 
     for name, value in os.environ.items():
         if name.startswith(prefix):
             config_key_name = name[len(prefix):]
+            if re.search('(?i)true|(?i)false', value):
```
Comment (atengler) [inline kqueen/config/utils.py]: If we are converting strings to bool, can't we convert json string to list/dict in python and avoid creating unnecessary class?

Comment (naumvd95) [reply kqueen/config/utils.py]: no, defining huge Json lists as Environment Variable for docker - is a bad case
variables should be simple and clear, all dict* stuff should be stored in configs

File: kqueen/config/auth.py (hunk 1 above)
Comment (naumvd95) [reply kqueen/config/auth.py]: to increase visibility and avoid case of multiple value-entries in config file, when 
a = 'One'
b = a + 'Two' == 'One' + 'Two'

apply_changes:
a == 'EnvOne'
b == 'One' + 'Two'

File: kqueen/auth/common.py [hunk 3]
```diff
@@ -12,18 +13,36 @@
 logger = logging.getLogger('kqueen_api')
 
 
+def generate_auth_config(auth_list):
+    auth_options = {}
+
+    methods = auth_list.split(',')
```
[the remark this answers is not in the document]
Comment (naumvd95) [reply kqueen/auth/common.py]: good point, thanks!

File: kqueen/auth/common.py [hunk 4]
```diff
@@ -12,18 +13,36 @@
 logger = logging.getLogger('kqueen_api')
 
 
+def generate_auth_config(auth_list):
```
[the remark this answers is not in the document]
Comment (naumvd95) [reply kqueen/auth/common.py]: okok

File: kqueen/config/auth.py (hunk 1 above)
Comment (naumvd95) [reply kqueen/config/auth.py]: we already discussed the case with u previously

Comment (naumvd95) [reply kqueen/config/auth.py]: and about class, i wanted to provide simple possibility to add another auth modules, if u have any other good solutions, please go ahead

File: kqueen/auth/common.py [hunk 5]
```diff
@@ -12,18 +13,36 @@
 logger = logging.getLogger('kqueen_api')
```
Comment (katyafervent) [inline kqueen/auth/common.py]: you can set  
AUTH_MAP =
 {'ldap' = {"engine": "LDAPAuth",
+            "param": {
+                "uri": config.get('LDAP_URI')
+            }
+            }
+    'local' = {"engine": "LocalAuth",
+             "param": {}
+             }
}

here and not to create a new class

**Pull request closed without merging**